src/pages/docs/tutorials/web-midi.md

```diff
@@ -0,0 +1,349 @@
+    const midiNote = event.note.number;
+    this.noteEmitter.emit("play", { midiNote });
+  });
+
+  // Add note off listener
+  this.selectedInput.addListener("noteoff", (event) => {
+    const midiNote = event.note.number;
+    this.noteEmitter.emit("stop", { midiNote });
+  });
+}
+```
+
+The `noteEmitter` sends note events that our audio implementation will receive. The events include:
+
+- `play`. Play a note.
+- `stop`. Stop a note.
+- `stopAll`. Stop all notes.
+
+When a new controller is selected, we stop all notes and remove event listeners from the previously selected input. Then, we look up the new controller by name and set it as the selected input. Lastly, we add `noteon` and `noteoff` event listeners to the selected input that will emit `play` and `stop` note events.
+
+## Audio
+
+The audio implementation includes an audio engine and a synth. The audio engine initializes an Elementary web renderer and connects it to a Web Audio context. The synth creates audio graphs representing our notes and passes them to the audio engine for rendering.
+
+### Engine
+
+The [engine module](https://github.com/bgins/elementary-webmidi-demo/blob/main/src/audio/engine.js) includes a `core` web renderer and a Web audio `context`.
+
+```JavaScript
+import WebRenderer from "@elemaudio/web-renderer";
+
+export class Engine {
+  context;
+  core;
+
+  constructor() {
+    this.core = new WebRenderer();
+    this.context = new AudioContext();
+    this.context.suspend();
+  }
+
+  // Class methods...
+}
+```
+
+Web Audio requires a user interaction before we can produce any sound, so we initially suspend the audio context.
+
+The `Engine` class has an `initialize` method that we will call on user interaction.
+
+```JavaScript
+async initialize() {
+  // Start the audio context
+  this.context.resume();
+
+  // Initialize web renderer
+  const node = await this.core.initialize(this.context, {
+    numberOfInputs: 0,
+    numberOfOutputs: 1,
+    outputChannelCount: [1],
+  });
+
+  // Connect web renderer to audio context
+  node.connect(this.context.destination);
+}
+```
+
+`initialize` resumes the audio context, initializes the `WebRenderer`, and connects it to the Web Audio context destination. The destination is conceptually your speakers, but it can be any device your browser connects to.
+
+We configure the WebRenderer with zero inputs, one output, and one output channel. Our app needs no inputs and outputs a mono signal.
+
+The `Engine` class also has a `render` method that renders an audio graph with an `ElemNode` type.
+
+```JavaScript
+render(node) {
+  if (this.context.state === "running") {
+    this.core.render(node);
+  }
+}
+```
+
+We check if the audio context is running and render the audio graph if so.
+
+### Synth
+
+The [synth module](https://github.com/bgins/elementary-webmidi-demo/blob/main/src/audio/synth.js) contains a few helper functions to create synth voices and silence and to combine synth voices into a polyphonic whole.
```

Comment:
nitpicky: I think we could just say "contains a few helper functions to create and sum polyphonic synth voices and generate silence"

Reply:
Agreed, that's clearer. 👍